### ISSUE TITLE
Bug: indoor/outdoor thermometer not working

### ISSUE THREAD

**Issue by Hk1020** [OpenWonderLabs/homebridge-switchbot#1024]
### Describe the Bug

I have an indoor/outdoor thermometer/hygrometer that I try to connect via BLE. I don’t have a hub. The temperature is shown correctly in the Switchbot app on an iPad but in homebridge it stays at 30°/0%. HB runs on a raspberry pi. The raspberry pi sees the thermometer just fine, hcitool lescan shows it. 

### To Reproduce

Install this plugin and configure indoor/outdoor thermometer. 

### Expected Behavior

Correct readings in the accessories. 

### Relevant Log Output

```shell
[8/14/2024, 7:10:19 PM] [SwitchBot] Initializing SwitchBot platform...
[8/14/2024, 7:10:19 PM] Loading 1 accessories...
[8/14/2024, 7:10:19 PM] [SwitchBot] Platform Config: {"maxRetries":3,"delayBetweenRetries":3}
[8/14/2024, 7:10:19 PM] [SwitchBot] Plugin Version: 3.7.0
[8/14/2024, 7:10:19 PM] [SwitchBot] Finished initializing platform: SwitchBot
[8/14/2024, 7:10:19 PM] [SwitchBot] Verifying Config
[8/14/2024, 7:10:19 PM] [SwitchBot] Config OK
[8/14/2024, 7:10:19 PM] [SwitchBot] setupBLE
[8/14/2024, 7:10:19 PM] [SwitchBot] Executed didFinishLaunching callback
[8/14/2024, 7:10:19 PM] [SwitchBot] SwitchBot Device Manual Config Set: [{"deviceId":"c7:6A:00:46:4F:19","configDeviceName":"thermohygro","configDeviceType":"WoIOSensor","connectionType":"BLE"}]
…                    
[8/14/2024, 7:10:19 PM] [SwitchBot] [DEBUG] Using debugMode Logging
[8/14/2024, 7:10:19 PM] [SwitchBot] [DEBUG] Using Default Max Retries
[8/14/2024, 7:10:19 PM] [SwitchBot] [DEBUG] Using Default Delay Between Retries
[8/14/2024, 7:10:19 PM] [SwitchBot] [DEBUG] Using Default Refresh Rate (2 minutes).
[8/14/2024, 7:10:19 PM] [SwitchBot] [DEBUG] Using Default Push Rate.
[8/14/2024, 7:10:19 PM] [SwitchBot] [DEBUG] Using Default Max Retries.
[8/14/2024, 7:10:19 PM] [SwitchBot] [DEBUG] Using Default Delay Between Retries.
[8/14/2024, 7:10:19 PM] [SwitchBot] Discovered WoIOSensor: c7:6A:00:46:4F:19
[8/14/2024, 7:10:19 PM] [SwitchBot] [DEBUG] Device: thermohygro registerDevice: true
[8/14/2024, 7:10:19 PM] [SwitchBot] [DEBUG] Device: thermohygro connectionType: BLE, will display in HomeKit
[8/14/2024, 7:10:19 PM] [SwitchBot] Scanning for BLE SwitchBot devices...
[8/14/2024, 7:10:19 PM] [SwitchBot] Adding new accessory: thermohygro deviceId: c7:6A:00:46:4F:19
[8/14/2024, 7:10:19 PM] [SwitchBot] WoIOSensor: thermohygro Using Debug Mode Logging: debugMode
[8/14/2024, 7:10:19 PM] [SwitchBot] WoIOSensor: thermohygro Using Platform Config refreshRate: 120, Default updateRate: 5, Platform Config pushRate: 0.1
[8/14/2024, 7:10:19 PM] [SwitchBot] WoIOSensor: thermohygro Using Default Max Retries: 5, Default Delay Between Retries: 3000
[8/14/2024, 7:10:19 PM] [SwitchBot] WoIOSensor: thermohygro Model: W3400010, BLE Model: w, BLE Model Name: WoIOSensorTH, BLE Model Friendly Name: Outdoor Meter
[8/14/2024, 7:10:19 PM] [SwitchBot] WoIOSensor: thermohygro Using Default scanDuration: 5
[8/14/2024, 7:10:19 PM] [SwitchBot] WoIOSensor: thermohygro Using Default MQTT URL: , Default mqttOptions: {}, Default mqttPubOptions: {}
[8/14/2024, 7:10:19 PM] [SwitchBot] WoIOSensor: thermohygro Retrieve initial values and update Homekit
[8/14/2024, 7:10:19 PM] [SwitchBot] WoIOSensor: thermohygro BLERefreshStatus
[8/14/2024, 7:10:19 PM] [SwitchBot] WoIOSensor: thermohygro Registering Webhook Event Handler
[8/14/2024, 7:10:19 PM] [SwitchBot] WoIOSensor: thermohygro is not listening webhook.
[8/14/2024, 7:10:19 PM] [SwitchBot] WoIOSensor uuid: c7:6A:00:46:4F:19-WoIOSensor, (b4c19f6e-a12c-4283-9c40-3274052c91d2)
[8/14/2024, 7:10:19 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Config: {"connectionType":"BLE"}
[8/14/2024, 7:10:19 PM] [SwitchBot] WoIOSensor: thermohygro Firmware Version: 3.7.0
[8/14/2024, 7:10:19 PM] [SwitchBot] WoIOSensor: thermohygro is listening to Platform BLE.
[8/14/2024, 7:10:19 PM] [SwitchBot] thermohygro External Accessory Mode: undefined
[8/14/2024, 7:10:19 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro version: 3.7.0
[8/14/2024, 7:10:19 PM] [SwitchBot] WoIOSensor: thermohygro bleMac: c7::6:a::00::4:6::4f::1:9
…

[8/14/2024, 7:12:19 PM] [SwitchBot] WoIOSensor: thermohygro BLERefreshStatus
[8/14/2024, 7:12:19 PM] [SwitchBot] WoIOSensor: thermohygro is listening to Platform BLE.
[8/14/2024, 7:12:19 PM] [SwitchBot] WoIOSensor: thermohygro bleMac: c7::6:a::00::4:6::4f::1:9
[8/14/2024, 7:14:19 PM] [SwitchBot] WoIOSensor: thermohygro BLERefreshStatus
[8/14/2024, 7:14:19 PM] [SwitchBot] WoIOSensor: thermohygro is listening to Platform BLE.
[8/14/2024, 7:14:19 PM] [SwitchBot] WoIOSensor: thermohygro bleMac: c7::6:a::00::4:6::4f::1:9
[8/14/2024, 7:16:19 PM] [SwitchBot] WoIOSensor: thermohygro BLERefreshStatus
[8/14/2024, 7:16:19 PM] [SwitchBot] WoIOSensor: thermohygro is listening to Platform BLE.
[8/14/2024, 7:16:19 PM] [SwitchBot] WoIOSensor: thermohygro bleMac: c7::6:a::00::4:6::4f::1:9
```

### Config for homebridge-switchbot

```shell
{
    "name": "SwitchBot",
    "credentials": {
        "notice": "Keep your Token & Secret a secret!"
    },
    "options": {
        "devices": [
            {
                "deviceId": "c7:6A:00:46:4F:19",
                "configDeviceName": "thermohygro",
                "configDeviceType": "WoIOSensor",
                "connectionType": "BLE"
            }
        ],
        "BLE": true
    },
    "platform": "SwitchBot"
}
```

### Device and Model

Indoor/outdoor thermometer 

### Node.js Version

v20.16.0

### NPM Version

10.2.3

### Homebridge Version

v1.8.4

### Homebridge Switchbot Plugin Version

v3.7.0

### Homebridge Config UI X Plugin Version

v4.56.4

### Operating System

Raspian/Debian

**Comment by donavanbecker** [comment]
that means that it isn't connecting to the meter

**Comment by Hk1020** [comment]
hm, and why does `hcitool` show it (last two lines):
```
raspberrypi> sudo hcitool lescan
LE Scan ...
74:F6:86:3D:DE:77 (unknown)
74:F6:86:3D:DE:77 (unknown)
5D:53:F2:4F:66:3B (unknown)
67:9B:FA:88:93:D9 (unknown)
67:9B:FA:88:93:D9 (unknown)
C7:6A:00:46:4F:19 (unknown)
C7:6A:00:46:4F:19 (unknown)
```

And the SB app:

![image](https://github.com/user-attachments/assets/931b0f05-6dc3-4f82-8a26-ad9de42220a6)

**Comment by donavanbecker** [comment]
just because it is showing it in a scan, doesn't mean that it is connecting to it.

**Comment by Hk1020** [comment]
ok, and how to make it connect?

**Comment by donavanbecker** [comment]
Get it closer to your homebrige device. I just tested out my own iosensor and was able to get that status.

**Comment by Hk1020** [comment]
thanks but no  change. it now sits right next to the raspi about 10cm from it. i also restarted homebridge. Must be something else.

**Comment by Hk1020** [comment]
And there’s also this
```
raspberrypi> sudo hcitool lescan
Set scan parameters failed: Input/output error
```

```
raspberrypi> bluetoothctl scan on
Discovery started
[CHG] Controller DC:A6:32:38:8C:22 Discovering: yes
…
[NEW] Device C7:6A:00:46:4F:19 C7-6A-00-46-4F-19
…        
[CHG] Device C7:6A:00:46:4F:19 ManufacturerData Key: 0x0969
[CHG] Device C7:6A:00:46:4F:19 ManufacturerData Value:
  c7 6a 00 46 4f 19 6d 0a 08 95 4b 00              .j.FO.m...K.    
[DEL] Device 54:E8:D5:87:39:38 54-E8-D5-87-39-38

```

But I am really familiar with blutooth configuration on the raspi.

**Comment by donavanbecker** [comment]
Can you change your deviceid from `c7:6A:00:46:4F:19` to `C76A00464F19`

**Comment by Hk1020** [comment]
This was it. It now works, thanks a lot. Maybe it would be possible to add a check to the plugin so it will reject or hint at a potential problem? The debug logs made me think everything is fine.

**Comment by donavanbecker** [comment]
valid. @Hk1020, I have added some better macaddress formatting validations to the latest beta.

Can you try the latest beta: 

[![npm version](https://badgen.net/badge/homebridge-switchbot@beta/>=3.8.2-beta.0/orange)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)

[Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)

**Comment by Hk1020** [comment]
Thanks, it kind of works. Just replacing the old version with this one and restarting HB worked as before. 

I then added colons to the MAC address which resulted in two devices. So I removed both from cache and restarted HB. I now have one device with apparently working humidity but broken temperature. 

Here’s the log
```
[9/14/2024, 12:01:02 PM] [SwitchBot] Stopped BLE scanning to close listening.
[9/14/2024, 12:01:20 PM] [SwitchBot] Initializing SwitchBot platform...
[9/14/2024, 12:01:20 PM] Loading 1 accessories...
[9/14/2024, 12:01:20 PM] [SwitchBot] Restoring existing accessory from cache: thermohygro deviceId: C7:6A:00:46:4F:19
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Using Device Config Logging: debug
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Using Platform Config refreshRate: 120, Default updateRate: 5, Platform Config pushRate: 0.1
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Using Default Max Retries: 5, Default Delay Between Retries: 3000
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Model: W3400010, BLE Model: w, BLE Model Name: WoIOSensorTH, BLE Model Friendly Name: undefined
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Using Default scanDuration: 5
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Using Default MQTT URL: , Default mqttOptions: {}, Default mqttPubOptions: {}
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Retrieve initial values and update Homekit
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro BLERefreshStatus
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Registering Webhook Event Handler
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro is not listening webhook.
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Registering Platform BLE Event Handler
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro registerPlatformBLE
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Config: {"logging":"debug","connectionType":"BLE","mode":"only_up"}
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Firmware Version: 3.8.2
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro is listening to Platform BLE.
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro version: 3.8.2
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro bleMac: c7:6a:00:46:4f:19
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro bleMac: c7:6a:00:46:4f:19
[9/14/2024, 12:01:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Scanning for WoIOSensorTH devices...
[9/14/2024, 12:01:30 PM] Homebridge v1.8.4 (HAP v0.12.2) (Homebridge 57BB) is running on port 51270.
[9/14/2024, 12:01:30 PM] 

NOTICE TO USERS AND PLUGIN DEVELOPERS
> Homebridge 2.0 is on the way and brings some breaking changes to existing plugins.
> Please visit the following link to learn more about the changes and how to prepare:
> https://github.com/homebridge/homebridge/wiki/Updating-To-Homebridge-v2.0

[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro received BLE: {"model":"w","modelName":"WoIOSensorTH","modelFriendlyName":"Outdoor Meter","temperature":{"c":11.9,"f":53.4},"celsius":11.9,"fahrenheit":53.4,"fahrenheit_mode":false,"humidity":69,"battery":100}
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro {
  "id": "c76a00464f19",
  "address": "c7:6a:00:46:4f:19",
  "rssi": -70,
  "serviceData": {
    "model": "w",
    "modelName": "WoIOSensorTH",
    "modelFriendlyName": "Outdoor Meter",
    "temperature": {
      "c": 11.9,
      "f": 53.4
    },
    "celsius": 11.9,
    "fahrenheit": 53.4,
    "fahrenheit_mode": false,
    "humidity": 69,
    "battery": 100
  }
}
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro address: c7:6a:00:46:4f:19, model: w
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro serviceData: {"model":"w","modelName":"WoIOSensorTH","modelFriendlyName":"Outdoor Meter","temperature":{"c":11.9,"f":53.4},"celsius":11.9,"fahrenheit":53.4,"fahrenheit_mode":false,"humidity":69,"battery":100}
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro BLEparseStatus
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro (battery, temperature, humidity) = BLE:(100, undefined, 69), current:(100, 30, 69)
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro BatteryLevel: 100
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro StatusLowBattery: 0
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro CurrentRelativeHumidity: 69%
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Temperature: undefined°c
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro updateCharacteristic CurrentRelativeHumidity: 69
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro CurrentRelativeHumidity context before: 69
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro CurrentRelativeHumidity context after: 69
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro CurrentTemperature: undefined
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro updateCharacteristic BatteryLevel: 100
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro BatteryLevel context before: 100
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro BatteryLevel context after: 100
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro updateCharacteristic StatusLowBattery: 0
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro StatusLowBattery context before: 0
[9/14/2024, 12:01:32 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro StatusLowBattery context after: 0
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro BLEparseStatus
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro (battery, temperature, humidity) = BLE:(100, undefined, 69), current:(100, undefined, 69)
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro BatteryLevel: 100
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro StatusLowBattery: 0
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro CurrentRelativeHumidity: 69%
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Temperature: undefined°c
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro updateCharacteristic CurrentRelativeHumidity: 69
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro CurrentRelativeHumidity context before: 69
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro CurrentRelativeHumidity context after: 69
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro CurrentTemperature: undefined
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro updateCharacteristic BatteryLevel: 100
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro BatteryLevel context before: 100
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro BatteryLevel context after: 100
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro updateCharacteristic StatusLowBattery: 0
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro StatusLowBattery context before: 0
[9/14/2024, 12:01:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro StatusLowBattery context after: 0
[9/14/2024, 12:03:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro BLERefreshStatus
[9/14/2024, 12:03:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro bleMac: c7:6a:00:46:4f:19
[9/14/2024, 12:03:30 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Scanning for WoIOSensorTH devices...
[9/14/2024, 12:03:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro BLEparseStatus
[9/14/2024, 12:03:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro (battery, temperature, humidity) = BLE:(undefined, undefined, undefined), current:(100, undefined, 69)
[9/14/2024, 12:03:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro BatteryLevel: undefined
[9/14/2024, 12:03:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro StatusLowBattery: 0
[9/14/2024, 12:03:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro CurrentRelativeHumidity: undefined%
[9/14/2024, 12:03:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro Temperature: undefined°c
[9/14/2024, 12:03:35 PM] [SwitchBot] [DEBUG] WoIOSensor: thermohygro CurrentRelativeHumidity: undefined
…
```

**Comment by donavanbecker** [comment]
I don't think that's related. Thanks for testing. If I was you, I would now remove the colons from your config.

**Comment by Hk1020** [comment]
I just did but the temperature stays undefined as per the logs. Seems to be reading it ok from the device but then messes up passing it on.

**Comment by azmke** [comment]
I can confirm this bug. I have a SwitchBot indoor/outdoor thermo-hygrometer. Battery level and humidity are read correctly and displayed in the Apple Home app. The temperature also seems to be read correctly at first (see 6.7 °C in the logs), but then for some reason it is converted (?) to `undefined` and displayed as 30 °C (default init value?) in the Apple Home app.

Here is another excerpt from the logs:

```

[9/19/2024, 1:28:12 AM] [SwitchBot] [DEBUG] WoIOSensor: ThermometerKuehlschrank {
  "id": "XXXXXXXXXXXX",
  "address": "xx:xx:xx:xx:xx:xx",
  "rssi": -84,
  "serviceData": {
    "model": "w",
    "modelName": "WoIOSensorTH",
    "modelFriendlyName": "Outdoor Meter",
    "temperature": {
      "c": 6.7,
      "f": 44.1
    },
    "celsius": 6.7,
    "fahrenheit": 44.1,
    "fahrenheit_mode": true,
    "humidity": 80,
    "battery": 88
  }
}
[9/19/2024, 1:28:12 AM] [SwitchBot] [DEBUG] WoIOSensor: ThermometerKuehlschrank address: xx:xx:xx:xx:xx:xx, model: w
[9/19/2024, 1:28:12 AM] [SwitchBot] [DEBUG] WoIOSensor: ThermometerKuehlschrank serviceData: {"model":"w","modelName":"WoIOSensorTH","modelFriendlyName":"Outdoor Meter","temperature":{"c":6.7,"f":44.1},"celsius":6.7,"fahrenheit":44.1,"fahrenheit_mode":true,"humidity":80,"battery":88}
[9/19/2024, 1:28:15 AM] [SwitchBot] [DEBUG] WoIOSensor: ThermometerKuehlschrank BLEparseStatus
[9/19/2024, 1:28:15 AM] [SwitchBot] [DEBUG] WoIOSensor: ThermometerKuehlschrank (battery, temperature, humidity) = BLE:(88, undefined, 80), current:(88, 30, 80)
[9/19/2024, 1:28:15 AM] [SwitchBot] [DEBUG] WoIOSensor: ThermometerKuehlschrank BatteryLevel: 88
[9/19/2024, 1:28:15 AM] [SwitchBot] [DEBUG] WoIOSensor: ThermometerKuehlschrank StatusLowBattery: 0
[9/19/2024, 1:28:15 AM] [SwitchBot] [DEBUG] WoIOSensor: ThermometerKuehlschrank CurrentRelativeHumidity: 80%
[9/19/2024, 1:28:15 AM] [SwitchBot] [DEBUG] WoIOSensor: ThermometerKuehlschrank Temperature: undefinedÂ°c
```

(Device ID and address have been anonymized for privacy reasons)

**Comment by azmke** [comment]
I think I've found the bug: Have a look at `src/device/iosensor.ts`. There the function `BLEparseStatus()` tries to access `this.serviceData.celcius` (with a **C** -> celCius). However, as you can see in the logs above, it should be called `this.serviceData.celsius` (with an **S** -> celSius). This is a typo. I have fixed this and opened a pull request.

**Comment by donavanbecker** [comment]
This is resolved with v3.8.2
